filecoin-proofs/src/api.rs

```diff
@@ -263,6 +333,44 @@ pub fn verify_seal(
     .map_err(Into::into)
 }
 
+pub fn verify_piece_inclusion_proof(
+    bytes: Vec<u8>,
+    comm_d: Commitment,
+    comm_p: Commitment,
+    piece_size: PaddedBytesAmount,
+    sector_size: SectorSize,
+) -> error::Result<bool> {
+    let piece_inclusion_proof: PieceInclusionProof<PedersenHasher> = bytes.into();
+    let comm_d = storage_proofs::hasher::pedersen::PedersenDomain::try_from_bytes(&comm_d)?;
+    let comm_p = storage_proofs::hasher::pedersen::PedersenDomain::try_from_bytes(&comm_p)?;
+    let piece_leaves = u64::from(piece_size) / 32;
+    let sector_leaves = u64::from(PaddedBytesAmount::from(sector_size)) / 32;
+
+    Ok(piece_inclusion_proof.verify(
+        &comm_d,
+        &comm_p,
+        piece_leaves as usize,
+        sector_leaves as usize,
+    ))
+}
+
+pub fn generate_piece_commitment<T: Into<PathBuf> + AsRef<Path>>(
+    unpadded_piece_path: T,
+    unpadded_piece_size: UnpaddedBytesAmount,
+) -> error::Result<(Commitment, PaddedBytesAmount)> {
+    let mut unpadded_piece_file = File::open(unpadded_piece_path)?;
+    let mut padded_piece_file = NamedTempFile::new().expects("could not create named temp file");
```

Comment:
Replace the `expects` with `?`. We don't want this thing to panic.

Reply:
Replace `NamedTempFile::new()` with `https://docs.rs/tempfile/3.1.0/tempfile/fn.tempfile.html`. You only need a `NamedTempFile` if you plan to pass its path around.